core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/dml.scala

```diff
@@ -38,13 +38,48 @@ case class UpdateTable(
   override def output: Seq[Attribute] = target.output
 }
 
-case class MergeTables(
+/**
+ * The logical plan of the MERGE INTO command, aligned with SparkSQL
+ */
+case class MergeIntoTable(
     target: LogicalPlan,
-    source: Option[LogicalPlan],
-    conditions: Option[Expression],
+    source: LogicalPlan,
+    mergeCondition: Expression,
+    matchedActions: Seq[MergeAction],
+    notMatchedActions: Seq[MergeAction],
+    notMatchedBySourceActions: Seq[MergeAction],
     outputRelation: Option[LogicalPlan],
     options: Option[Expression])
     extends Modification {
-  override def children: Seq[LogicalPlan] = Seq(target, source.getOrElse(NoopNode), outputRelation.getOrElse(NoopNode))
+
+  override def children: Seq[LogicalPlan] = Seq(target, source, outputRelation.getOrElse(NoopNode))
```

Comment:
don't add `NoopNode`, but do just the `++ outputRelation`. also, looking at `InsertAction`, you may want to add logical plans from there to `children` here

Reply:
Corrected and InsertClause values added to children